src/main/java/org/hydev/mcpm/utils/arrays/SlidingWindow.java

```diff
@@ -0,0 +1,55 @@
+package org.hydev.mcpm.utils.arrays;
+
+import org.hydev.mcpm.utils.Pair;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+/**
+ * An implementation of FixedWindowSum using a queue.
+ *
+ * @author Peter (https://github.com/MstrPikachu)
+ * @since 2022-11-21
+ */
+public class SlidingWindow implements FixedWindowSum {
+    private long sum;
+    private long window;
+    private final Queue<Pair<Long, Long>> queue = new ArrayDeque<>();
+
+    @Override
+    public FixedWindowSum setWindowSize(long window) {
+        this.window = window;
+        return this;
+    }
+
+    @Override
+    public void add(long index, long inc) {
+        queue.add(new Pair<>(index, inc));
+        sum += inc; // increase sum after adding a new value
+        restore(index);
+    }
+
+    @Override
+    public long sum() {
+        return sum;
+    }
+
+    @Override
+    public long sum(long index) {
+        restore(index);
+        return sum;
+    }
```

Comment:
Not that I dislike this but overloading to me implies that you have similar behaviour (especially when implementing methods in an interface). I would maybe suggest going with the traditional `getSum`/`setSum` combo.

Reply:
I agree but to me the behaviour does seem similar, `sum` does the same as `sum(index)` with the most recent index. Also `setSum` shouldn't be allowed because it's private.

Maybe you're suggesting that I violated the interface segregation principle which is probably true if I look at it again, but then it would be an issue about the design and not the tests anymore. I can fix/make an issue for that afterwards